internal/pkg/api/server/templates/service/service.go

```diff
@@ -237,8 +238,68 @@ func (s *service) UpdateInstance(dependencies.Container, *UpdateInstancePayload)
 	return nil, NotImplementedError{}
 }
 
-func (s *service) DeleteInstance(dependencies.Container, *DeleteInstancePayload) (err error) {
-	return NotImplementedError{}
+func (s *service) DeleteInstance(d dependencies.Container, payload *DeleteInstancePayload) (err error) {
+	// Note:
+	//   Waits for separation of remote and local state.
+	//   A virtual FS and fake manifest are created to make it work.
+
+	branchKey, err := getBranch(d, payload.Branch)
+	if err != nil {
+		return err
+	}
+
+	// Create virtual fs, after refactoring it will be removed
+	fs, err := aferofs.NewMemoryFs(d.Logger(), "")
+	if err != nil {
+		return err
+	}
+
+	// Create fake manifest
+	m := project.NewManifest(123, "foo")
+
+	// Load only target branch
+	m.Filter().SetAllowedKeys([]model.Key{branchKey})
+	prj := project.NewWithManifest(fs, m, d)
+
+	// Load project state
+	prjState, err := prj.LoadState(loadState.Options{LoadRemoteState: true})
+	if err != nil {
+		return err
+	}
+
+	// Copy remote state to the local
+	for _, objectState := range prjState.All() {
+		objectState.SetLocalState(deepcopy.Copy(objectState.RemoteState()).(model.Object))
+	}
```

Comment:
@webrouse prosím tě, kopíruju tady ten "hack" na load stavu projektu z `UseTemplateVersion()` a to má teda pullnout všechny konfigy té jedné branche? V `prjState.All()` mám jen tu branch a nic jinýho, tak přemýšlím kde jsem udělal chybu.

Reply:
Pri aplikovani sablony, nepotrebujeme ostatne konfiguracie (kedze sa iba vytvaraju).
Pri mazani potrebujes konfiguracie, ... preto je potrebne upravit filter ... namiesto `.SetAllowedKeys([]model.Key{branchKey})` pouzi `SetAllowedBranches` ... tj. nestiahne sa iba branch, ale vsetko z branche. 

Viem, ze je to take ne-intuitívne, ... bude to lepsie ked sa toho hacku zbavime.

Reply:
Jo, díky. Už to vidím u těch ostatních operací, jasný. 👍